journey-map/public/index.mjs

```diff
@@ -113,6 +113,45 @@ const resolveEventTarget = (definition, formData) => {
     return definition.targetState;
 };
 
+const resolveAllPossibleEventTargets = (eventDefinition) => {
+    let targets = [];
+    targets.push(eventDefinition.targetState);
+    if (eventDefinition.checkIfDisabled) {
+        Object.values(eventDefinition.checkIfDisabled).forEach((def) => {
+            targets.push(...resolveAllPossibleEventTargets(def));
+        })
+    }
+
+    if (eventDefinition.checkFeatureFlag) {
+        Object.values(eventDefinition.checkFeatureFlag).forEach((def) => {
+            targets.push(...resolveAllPossibleEventTargets(def));
+        })
+    }
+
+    return targets;
+};
+
+const calcOrphanStates = (journeyMap) => {
+    const targetedStates = [...initialStates]
+
+    Object.values(journeyMap).forEach((stateDefinition) => {
+        if (stateDefinition.events) {
+            Object.values(stateDefinition.events).forEach((eventDefinition) => {
+                targetedStates.push(...resolveAllPossibleEventTargets(eventDefinition));
+            })
+        }
+        if (stateDefinition.exitEvents) {
+            Object.values(stateDefinition.exitEvents).forEach((eventDefinition) => {
+                targetedStates.push(...resolveAllPossibleEventTargets(eventDefinition));
+            })
+        }
+    })
+
+    const uniqueTargetedStates = [...new Set(targetedStates)];
```

Comment:
Extending the idea above, you could do this in one go as well.
```
const targetedStates = [
    ...initialStates,
    ...Object.values(journeyMap).flatMap((stateDefinition) => [
        ...Object.values(stateDefinition.events || {}).flatMap(resolveAllPossibleEventTargets),
        ...Object.values(stateDefinition.exitEvents || {}).flatMap(resolveAllPossibleEventTargets),
    ]),
];
```
(You could even inline the `Set`, but that feels like a distinct thing)

Reply:
Using functional methods effectively is one of the things I love doing :D - and one reason I like TS, as it works pretty well for it, much better than Java streams.

(My favourite language used to be Scala, which is even better :P - but haven't written any in years)

Reply:
I only have one experience with Scala and that was years ago writing Gatling tests. And it burned me.